src/Desktop/UIAutomation/CustomObjects/CustomProperty.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+
+namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
+{
+    public class CustomProperty
+    {
+        /// <summary>The RFC4122 globally unique identifier of this property.</summary>
+        [JsonProperty("guid")]
+        #pragma warning disable CA1720 // Identifier contains type name: name from JSON
+        public Guid? Guid { get; set; }
+        #pragma warning restore CA1720 // Identifier contains type name: name from JSON
+        /// <summary>A textual description of this property.</summary>
+        [JsonProperty("programmaticName")]
+        public string ProgrammaticName { get; set; }
```

Comment:
Let's don't overthink this: this property is just a "Name". "Programmatic" doesn't add distinguishing information.

Reply:
How much do we care about [being consistent with UIA](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcore/ns-uiautomationcore-uiautomationpropertyinfo) here? (CTRL+f for "programmatic").